src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -596,18 +596,19 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
                     Vector128<byte> values = Vector128.Create(value);
                     while (lengthToExamine > offset)
                     {
-                        Vector128<byte> search = LoadVector128(ref searchSpace, offset);
+                        Vector128<byte> search = Vector128.LoadUnsafe(ref searchSpace, offset);
 
                         // Same method as above
-                        int matches = Sse2.MoveMask(Sse2.CompareEqual(values, search));
-                        if (matches == 0)
+                        Vector128<byte> compareResult = Vector128.Equals(values, search);
+                        if (compareResult == Vector128<byte>.Zero)
                         {
                             // Zero flags set so no matches
                             offset += (nuint)Vector128<byte>.Count;
                             continue;
                         }
 
                         // Find bitflag offset of first match and add to current offset
+                        uint matches = compareResult.ExtractMostSignificantBits();
```

Comment:
Last time I tried to do the same it was a noticeable regression for ARM64

Reply:
Same here, but https://github.com/dotnet/runtime/pull/73364/commits/28aa08214b44042c5c14d85bc5512820f775d040 solved the problem

Reply:
Now this operation is performed only once, after we find a match (not once for every vector we compare)

Reply:
What is the cost of this approach vs doing it every loop for `Vector256<T>`

Is it better, particularly for large inputs, to do this there as well?